src/lxc/start.c

```diff
@@ -344,12 +344,36 @@ static int signal_handler(int fd, uint32_t events, void *data,
 int lxc_set_state(const char *name, struct lxc_handler *handler,
 		  lxc_state_t state)
 {
+	int state_socket;
 	ssize_t ret;
 	struct lxc_list *cur, *next;
 	struct state_client *client;
 	struct lxc_msg msg = {.type = lxc_msg_state, .value = state};
 
+	if (state != STARTING) {
+		state_socket = lxc_abstract_unix_connect(handler->state_socket_name);
+		if (state_socket < 0) {
+			if (errno == ECONNREFUSED)
```

Comment:
This means `wait_on_daemonized_start()` has closed the connection so we won't bother trying to send. Instead, we go right to serving any registered state clients.